jonnyandrew/synth
# Add unit tests for the C++ code

# Description
Add the framework for unit testing the C++ code in the native library and a few unit tests to demonstrate it working.

The tests are built to run on an Android device or emulator. CMake also generates a script that pushes the test binary and libraries to the device and then runs them.

This script can be found in the CMake binary directory:
```
app/.cxx/cmake/<build>/<abi>/cpp_test_runner.sh
```

## How to run the tests
Lets say we want to run the tests on an x86 emulator.
- Ensure a device or emulator is connected using ADB:
```
adb devices
```
- Open a terminal and change to the project directory
- Build the project
```
./gradlew assembleDebug
```
- Now run the following:
```
chmod +x ./app/.cxx/cmake/debug/x86/cpp_test_runner.sh
./app/.cxx/cmake/debug/x86/cpp_test_runner.sh
```
